src/maintainer/infrastructure.rst

```diff
@@ -281,35 +281,112 @@ Compilers and Runtimes
 
 Conda-forge builds and maintains its own set of compilers for various languages
 and/or systems (e.g., ``C``, ``FORTRAN``, ``C++``, ``CUDA``, etc.). These are used
-in all of our CI builds to build both core dependencies (e.g., ``Python``) and maintainer-contributed
-packages. While we do not have any formal policies or promises of support for these
+in all of our CI builds to build essentially all artefacts published by conda-forge.
+
+In the past, compiler upgrades often required full rebuilds of basically all of
+conda-forge due to potential ABI breaks. These breaks have become much rarer in
+frequency and scope, so it is not a current concern. However, we keep our policies
+for such ABI breaks in place for the next time it should occur.
+
+While we do not have any promises of support for a generation of ABI-compatible
 compilers, we have historically maintained them according to the following (non-binding)
 principles.
 
 * The authoritative source of the current compilers and versions for various languages
   and platforms is the `conda_build_config.yaml <https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml>`_
   in the `conda-forge/conda-forge-pinning-feedstock <https://github.com/conda-forge/conda-forge-pinning-feedstock>`_
   as described in :ref:`globally_pinned_packages`.
-* We provide no support of any kind in terms of the long-term stability of these pinnings.
+* We provide no support of any kind in terms of the long-term stability / support of a given compiler generation.
 * We upgrade them in an ad-hoc manner on a periodic basis as we have the time and energy to do so.
   Note that because of the way we enforce runtime constraints, these compiler upgrades will not break
   existing packages. However, if you are using the compilers outside of ``conda``, then you may find issues.
-* We generally provide notice in the form of an announcement when a compiler is going to be upgraded.
+* We usually provide notice in the form of an announcement when an ABI-incompatible compiler change is going to happen.
```

Comment:
Can you please keep this as it was? It was agreed upon in a core meeting and I don't want to change it.

Reply:
Yes will do. My apologies. Not right this minute though.

Reply:
Done. https://github.com/conda-forge/conda-forge.github.io/pull/1947

Reply:
> Done. https://github.com/conda-forge/conda-forge.github.io/pull/1947

We're not in such a rush IMO... I didn't press merge here, I'd be happy to discuss this some more. Aside from a typo ("conpiler"), #1947 is IMO not correct, because that section now explicitly tries to draw a distinction between ABI-breaking and ABI-preserving compiler upgrades, and you made that distinction less clear.

Reply:
I'll fix the typo ofc but I'm not gonna split hairs on the language here. Isuru is correct that this language was agreed upon in a core meeting and so should remain unchanged. I mentioned this in the previous pr as well at some point iirc. 

Please accept my apologies for merging too early. The pr looked good to go and so I went ahead. 

I'm happy to back out the prs or we can continue debate in a new one. 

I really appreciate the hard work from you @h-vetinari on this. 

One thing I realized is that we don't separate out what is an informal policy from core and what is more general documentation. That is making this process more difficult than it has to be.

Reply:
> this language was agreed upon in a core meeting and so should remain unchanged

I disagree with the tenet that something that has been (core-)agreed once is forever unchangeable, especially if circumstances change.

You can back out my PR (or not 🤷); my interest here is not to offend but to provide useful documentation, in particular the distinction between ABI-breaking and not ABI-breaking compiler upgrades[^1]. If that needs a core vote, then that's fine by me, but I wrote this text specifically with that separation in mind. If people don't want that separation, just revert.

[^1]: as an instance of the latter is what started this when John [asked](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4215#issuecomment-1518358829) for a docs update